New batch of warnings

## Issue Details

Here is a list of recently appeared warnings in the testsuite :

 - [ ] https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.2-I-14/Boolean_set_operations_2_Examples/TestReport_lrineau_Ubuntu-latest-GCC6-Release.gz :  /mnt/testsuite/include/CGAL/Arr_geometry_traits/Bezier_x_monotone_2.h:249:54: warning: array subscript 'const CGAL::_Bezier_x_monotone_2<CGAL::Cartesian<CORE::BigRat>, CGAL::Cartesian<CORE::Expr>, CGAL::CORE_algebraic_number_traits, CGAL::Bezier_bounding_rational_traits<CGAL::Cartesian<CORE::BigRat> > >[0]' is partly outside array bounds of 'Polygon_2 [1]' {aka 'CGAL::General_polygon_2<CGAL::Arr_Bezier_curve_traits_2<CGAL::Cartesian<CORE::BigRat>, CGAL::Cartesian<CORE::Expr>, CGAL::CORE_algebraic_number_traits> > [1]'} [-Warray-bounds]
  249 |   bool is_directed_right() const { return (_dir_right); }

 - [x] https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.2-I-14/Convex_decomposition_3/TestReport_lrineau_Ubuntu-latest-GCC6.gz : /mnt/testsuite/include/CGAL/Nef_2/debug.h:61:28: warning: this 'if' clause does not guard... [-Wmisleading-indentation]
   61 | #define CGAL_NEF_TRACEN(t) if((debugthread%CGAL_NEF_DEBUG)==0) 

 - [x] https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.2-I-14/Hash_map/TestReport_lrineau_Ubuntu-latest-GCC6-Release.gz : /mnt/testsuite/include/CGAL/Triangulation_2.h:1354:16: warning: 'li' may be used uninitialized [-Wmaybe-uninitialized]
 1354 |   return insert(p, lt, loc, li);
      |          ~~~~~~^~~~~~~~~~~~~~~~
/mnt/testsuite/include/CGAL/Triangulation_2.h:1352:7: note: 'li' declared here
 1352 |   int li;

 - [x] https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.2-I-14/Kernel_23/TestReport_lrineau_Ubuntu-latest-GCC6-Release.gz : /mnt/testsuite/include/CGAL/Intersections_2/Line_2_Ray_2.h:76:10: warning: 'ispair.CGAL::Intersections::internal::Ray_2_Line_2_pair<CGAL::Simple_cartesian<double> >::_intersection_point.CGAL::Point_2<CGAL::Simple_cartesian<double> >::<anonymous>.CGAL::PointC2<CGAL::Simple_cartesian<double> >::base.CGAL::Vector_2<CGAL::Simple_cartesian<double> >::<anonymous>.CGAL::VectorC2<CGAL::Simple_cartesian<double> >::base.std::array<double, 2>::_M_elems[1]' may be used uninitialized in this function [-Wmaybe-uninitialized]
   76 |     is_t ispair(&ray, &line);

- [ ] https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.2-I-14/Number_types/TestReport_lrineau_Ubuntu-latest-GCC6-Release.gz : mnt/testsuite/include/CGAL/Quotient.h:456:22: warning: '<anonymous>' may be used uninitialized [-Wmaybe-uninitialized]
  456 |   return quotient_cmp(x,y) == SMALLER;

- [x] https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.2-Ic-18/Tetrahedral_remeshing/TestReport_lrineau_Ubuntu-latest-GCC6.gz : CGAL/Tetrahedral_remeshing/internal/smooth_vertices.h:546:42: warning: loop variable 'si' creates a copy from type 'const Surface_patch_index' {aka 'const std::pair<int, int>'} 

- [ ] https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.2-Ic-18/CGAL_ipelets_Demo/TestReport_lrineau_Debian-stable-Release.gz : /include/CGAL/Arr_segment_traits_2.h:78:9: warning: '*((void*)(&<anonymous>)+8).CGAL::Arr_segment_traits_2<CGAL::Epeck>::_Segment_cached_2::m_is_directed_right' may be used uninitialized in this function 

- [x] https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.0.4-I-238/Polygon_mesh_processing/TestReport_lrineau_Ubuntu-latest-GCC6-CXX1z.gz: loop variable 'hk' creates a copy from type 'const halfedges_pair' {aka 'const std::pair<CGAL::SM_Halfedge_index, CGAL::SM_Halfedge_index>'} [-Wrange-loop-construct]

- [x] https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.1.1-I-243/Tetrahedral_remeshing/TestReport_lrineau_Ubuntu-latest-GCC6-Release.gz : /mnt/testsuite/include/CGAL/Tetrahedral_remeshing/internal/flip_edges.h:1104:25: warning: loop variable 'vp' creates a copy from type 'const std::pair<CGAL::internal::CC_iterator<CGAL::Compact_container<CGAL::Mesh_vertex_base_3<CGAL::Robust_weighted_circumcenter_filtered_traits_3<CGAL::Epick>, CGAL::Labeled_mesh_domain_3<CGAL::Epick>, CGAL::Regular_triangulation_vertex_base_3<CGAL::Robust_weighted_circumcenter_filtered_traits_3<CGAL::Epick>, CGAL::Triangulation_ds_vertex_base_3<CGAL::Triangulation_data_structure_3<CGAL::Mesh_vertex_base_3<CGAL::Robust_weighted_circumcenter_filtered_traits_3<CGAL::Epick>, CGAL::Labeled_mesh_domain_3<CGAL::Epick>, CGAL::Regular_triangulation_vertex_base_3<CGAL::Robust_weighted_circumcenter_filtered_traits_3<CGAL::Epick>, CGAL::Triangulation_ds_vertex_base_3<void> > >, CGAL::Compact_mesh_cell_base_3<CGAL::Robust_weighted_circumcenter_filtered_traits_3<CGAL::Epick>, CGAL::Labeled_mesh_domain_3<CGAL::Epick>, void>, CGAL::Sequential_tag> > > >, CGAL::Default, CGAL::Default, CGAL::Default>, false>, CGAL::internal::CC_iterator<CGAL::Compact_container<CGAL::Mesh_vertex_base_3<CGAL::Robust_weighted_circumcenter_filtered_traits_3<CGAL::Epick>, CGAL::Labeled_mesh_domain_3<CGAL::Epick>, CGAL::Regular_triangulation_vertex_base_3<CGAL::Robust_weighted_circumcenter_filtered_traits_3<CGAL::Epick>, CGAL::Triangulation_ds_vertex_base_3<CGAL::Triangulation_data_structure_3<CGAL::Mesh_vertex_base_3<CGAL::Robust_weighted_circumcenter_filtered_traits_3<CGAL::Epick>, CGAL::Labeled_mesh_domain_3<CGAL::Epick>, CGAL::Regular_triangulation_vertex_base_3<CGAL::Robust_weighted_circumcenter_filtered_traits_3<CGAL::Epick>, CGAL::Triangulation_ds_vertex_base_3<void> > >, CGAL::Compact_mesh_cell_base_3<CGAL::Robust_weighted_circumcenter_filtered_traits_3<CGAL::Epick>, CGAL::Labeled_mesh_domain_3<CGAL::Epick>, void>, CGAL::Sequential_tag> > > >, CGAL::Default, CGAL::Default, CGAL::Default>, false> >' [-Wrange-loop-construct]
 1104 |   for (const VertexPair vp : edges)

- [ ] https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.2-I-19/Arrangement_on_surface_2/TestReport_Friedrich_Ubuntu-gcc-7.gz: include/CGAL/Arr_geometry_traits/Curve_data_aux.h:54:7: warning: ‘*((void*)(&<anonymous>)+8).CGAL::_Curve_data_ex<CGAL::internal::X_monotone_polycurve_2<CGAL::Arr_segment_2<CGAL::Cartesian<CGAL::Gmpq> >, CGAL::Point_2<CGAL::Cartesian<CGAL::Gmpq> > >, unsigned int>::m_data’ may be used uninitialized in this function [-Wmaybe-uninitialized]
 class _Curve_data_ex : public BaseCurveType

- [ ] https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.2-I-19/Arrangement_on_surface_2_Examples/TestReport_lrineau_Debian-stable-Release.gz: More maybe unintialized

- [ ] https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.2-I-19/CGAL_ipelets_Demo/TestReport_lrineau_Debian-stable-Release.gz

- [x] Master : doc problem in Buunding volumes: Bounding_volumes/doc/Bounding_volumes/CGAL/Min_sphere_d.h:66: warning: unable to resolve reference to `Min_sphere_d/min_sphere_homogeneous_d.cpp' for \ref command
/mnt/cgal/Bounding_volumes/doc/Bounding_volumes/CGAL/Min_sphere_d.h:66: warning: included file Min_sphere_d/min_sphere_homogeneous_d.cpp is not found. Check your EXAMPLE_PATH

- [x] Master: In file included from /home/cgal_tester/build/src/cmake/platforms/ArchLinux-CXX14/test/Polyhedron_IO_Examples/off2off.cpp:4:
/mnt/testsuite/include/CGAL/IO/File_writer_OFF.h: In member function ‘void CGAL::File_writer_OFF::write_facet_begin(std::size_t)’:
/mnt/testsuite/include/CGAL/IO/File_writer_OFF.h:80:74: error: ‘int32_t’ in namespace ‘boost’ does not name a type
   80 |           I_Binary_write_big_endian_integer32( out(), static_cast<boost::int32_t>(no));
      |                                                                          ^~~~~~~
/mnt/testsuite/include/CGAL/IO/File_writer_OFF.h: In member function ‘void CGAL::File_writer_OFF::write_facet_vertex_index(std::size_t)’:
/mnt/testsuite/include/CGAL/IO/File_writer_OFF.h:86:74: error: ‘int32_t’ in namespace ‘boost’ does not name a type
   86 |           I_Binary_write_big_endian_integer32( out(), static_cast<boost::int32_t>(index));
**We should use std::int32_t instead, and incllude <cstdint> in that file**

--- a/Intersections_2/include/CGAL/Intersections_2/Line_2_Line_2.h
+++ b/Intersections_2/include/CGAL/Intersections_2/Line_2_Line_2.h
@@ -36,10 +36,10 @@ namespace internal {
 template <class K>
 class Line_2_Line_2_pair {
 public:
-    enum Intersection_results {NO_INTERSECTION, POINT, LINE};
+    enum Intersection_results {NOT_COMPUTED_YET, NO_INTERSECTION, POINT, LINE};
     Line_2_Line_2_pair(typename K::Line_2 const *line1,
                        typename K::Line_2 const *line2)
-      : _line1(line1), _line2(line2), _known(false) {}
+      : _line1(line1), _line2(line2), _result(NOT_COMPUTED_YET) {}
 
     Intersection_results intersection_type() const;
 
@@ -48,7 +48,6 @@ public:
 protected:
     typename K::Line_2 const*   _line1;
     typename K::Line_2 const *  _line2;
-    mutable bool                    _known;
     mutable Intersection_results    _result;
     mutable typename K::Point_2         _intersection_point;
 };
@@ -141,11 +140,10 @@ typename Line_2_Line_2_pair<K>::Intersection_results
 Line_2_Line_2_pair<K>::intersection_type() const
 {
     typedef typename K::RT RT;
-    if (_known)
+    if (_result != NOT_COMPUTED_YET)
         return _result;
     RT nom1, nom2, denom;
     // The non const this pointer is used to cast away const.
-    _known = true;
     denom = _line1->a()*_line2->b() - _line2->a()*_line1->b();
     if (denom == RT(0)) {
         if (RT(0) == (_line1->a()*_line2->c() - _line2->a()*_line1->c()) &&
@@ -180,7 +178,7 @@ template <class K>
 typename K::Point_2
 Line_2_Line_2_pair<K>::intersection_point() const
 {
-    if (!_known)
+    if (_result == NOT_COMPUTED_YET)
         intersection_type();
     CGAL_kernel_assertion(_result == POINT);
     return _intersection_point;
@@ -190,7 +188,7 @@ template <class K>
 typename K::Line_2
 Line_2_Line_2_pair<K>::intersection_line() const
 {
-    if (!_known)
+    if (_result == NOT_COMPUTED_YET)
         intersection_type();
     CGAL_kernel_assertion(_result == LINE);
     return *_line1;

--- a/Intersections_2/include/CGAL/Intersections_2/Line_2_Ray_2.h
+++ b/Intersections_2/include/CGAL/Intersections_2/Line_2_Ray_2.h
@@ -124,6 +124,7 @@ Ray_2_Line_2_pair<K>::intersection_type() const
     Line_2_Line_2_pair<K> linepair(&l1, _line);
     switch ( linepair.intersection_type()) {
     case Line_2_Line_2_pair<K>::NO_INTERSECTION:
+    default:
         _result = NO_INTERSECTION;
         break;
     case Line_2_Line_2_pair<K>::POINT:

--- a/Intersections_2/include/CGAL/Intersections_2/Line_2_Ray_2.h
+++ b/Intersections_2/include/CGAL/Intersections_2/Line_2_Ray_2.h
@@ -35,10 +35,13 @@ namespace internal {
 template <class K>
 class Ray_2_Line_2_pair {
 public:
-    enum Intersection_results {NO_INTERSECTION, POINT, RAY};
+    enum Intersection_results {NOT_COMPUTED_YET, NO_INTERSECTION, POINT, RAY};
+    typedef typename K::FT FT;
     Ray_2_Line_2_pair(typename K::Ray_2 const *ray,
                       typename K::Line_2 const *line)
-      : _ray(ray), _line(line), _known(false) {}
+      : _ray(ray), _line(line), _result(NOT_COMPUTED_YET),
+        _intersection_point(typename K::Point_2(FT(0), FT(0)))
+    {}
 
     Intersection_results intersection_type() const;
 
@@ -47,7 +50,6 @@ public:
 protected:
     typename K::Ray_2 const *   _ray;
     typename K::Line_2 const *  _line;
-    mutable bool                    _known;
     mutable Intersection_results    _result;
     mutable typename K::Point_2         _intersection_point;
 };
@@ -94,7 +96,7 @@ intersection(const typename K::Line_2 &line,
              const typename K::Ray_2 &ray,
              const K& k)
 {
-  return internal::intersection(ray, line, k);
+    return internal::intersection(ray, line, k);
 }
 
 
@@ -115,10 +117,9 @@ template <class K>
 typename Ray_2_Line_2_pair<K>::Intersection_results
 Ray_2_Line_2_pair<K>::intersection_type() const
 {
-    if (_known)
+    if (_result != NOT_COMPUTED_YET)
         return _result;
     // The non const this pointer is used to cast away const.
-    _known = true;
     const typename K::Line_2 &l1 = _ray->supporting_line();
     Line_2_Line_2_pair<K> linepair(&l1, _line);
     switch ( linepair.intersection_type()) {
@@ -142,7 +143,7 @@ template <class K>
 typename K::Point_2
 Ray_2_Line_2_pair<K>::intersection_point() const
 {
-    if (!_known)
+    if (_result == NOT_COMPUTED_YET)
         intersection_type();
     CGAL_kernel_assertion(_result == POINT);
     return _intersection_point;
@@ -152,7 +153,7 @@ template <class K>
 typename K::Ray_2
 Ray_2_Line_2_pair<K>::intersection_ray() const
 {
-    if (!_known)
+    if (_result == NOT_COMPUTED_YET)
         intersection_type();
     CGAL_kernel_assertion(_result == RAY);
     return *_ray;

--- a/Intersections_2/include/CGAL/Intersections_2/Line_2_Ray_2.h
+++ b/Intersections_2/include/CGAL/Intersections_2/Line_2_Ray_2.h
@@ -40,7 +40,7 @@ public:
     Ray_2_Line_2_pair(typename K::Ray_2 const *ray,
                       typename K::Line_2 const *line)
       : _ray(ray), _line(line), _result(NOT_COMPUTED_YET),
-        _intersection_point(typename K::Point_2(ORIGIN))
+        _intersection_point(K().construct_point_2_object()(ORIGIN))
     {}
 
     Intersection_results intersection_type() const;

--- a/Intersections_2/include/CGAL/Intersections_2/Line_2_Ray_2.h
+++ b/Intersections_2/include/CGAL/Intersections_2/Line_2_Ray_2.h
@@ -40,7 +40,7 @@ public:
     Ray_2_Line_2_pair(typename K::Ray_2 const *ray,
                       typename K::Line_2 const *line)
       : _ray(ray), _line(line), _result(NOT_COMPUTED_YET),
-        _intersection_point(ORIGIN)
+        _intersection_point(typename K::Point_2(ORIGIN))
     {}
 
     Intersection_results intersection_type() const;

--- a/Intersections_2/include/CGAL/Intersections_2/Line_2_Ray_2.h
+++ b/Intersections_2/include/CGAL/Intersections_2/Line_2_Ray_2.h
@@ -40,7 +40,7 @@ public:
     Ray_2_Line_2_pair(typename K::Ray_2 const *ray,
                       typename K::Line_2 const *line)
       : _ray(ray), _line(line), _result(NOT_COMPUTED_YET),
-        _intersection_point(typename K::Point_2(FT(0), FT(0)))
+        _intersection_point(ORIGIN)
     {}
 
     Intersection_results intersection_type() const;

--- a/Intersections_2/include/CGAL/Intersections_2/Segment_2_Segment_2.h
+++ b/Intersections_2/include/CGAL/Intersections_2/Segment_2_Segment_2.h
@@ -308,6 +308,7 @@ Segment_2_Segment_2_pair<K>::intersection_type() const
     Line_2_Line_2_pair<K> linepair(&l1, &l2);
     switch ( linepair.intersection_type()) {
     case Line_2_Line_2_pair<K>::NO_INTERSECTION:
+    default:
         _result = NO_INTERSECTION;
         break;
     case Line_2_Line_2_pair<K>::POINT:

--- a/Intersections_2/include/CGAL/Intersections_2/internal/Straight_2.h
+++ b/Intersections_2/include/CGAL/Intersections_2/internal/Straight_2.h
@@ -275,6 +275,7 @@ cut_right_off(typename K::Line_2 const & cutter)
     Line_2_Line_2_pair<K> pair(&support_, &cutter);
     switch (pair.intersection_type()) {
     case Line_2_Line_2_pair<K>::NO_INTERSECTION:
+    default:
         if (cutter.has_on_negative_side(support_.point()))
             bound_state_ = LINE_EMPTY;
         break;

--- a/Kernel_23/examples/Kernel_23/MyPointC2.h
+++ b/Kernel_23/examples/Kernel_23/MyPointC2.h
@@ -21,6 +21,13 @@ public:
     *(vec+1) = 0;
   }
 
+  MyPointC2(const CGAL::Origin &)
+    :col(0)
+  {
+    *vec = 0;
+    *(vec+1) = 0;
+  }
+
 
   MyPointC2(const double x, const double y, int c = 0)
     : col(c)

--- a/Kernel_23/examples/Kernel_23/MyPointC2.h
+++ b/Kernel_23/examples/Kernel_23/MyPointC2.h
@@ -21,14 +21,6 @@ public:
     *(vec+1) = 0;
   }
 
-  MyPointC2(const CGAL::Origin &)
-    :col(0)
-  {
-    *vec = 0;
-    *(vec+1) = 0;
-  }
-
-
   MyPointC2(const double x, const double y, int c = 0)
     : col(c)
   {

--- a/Triangulation_2/include/CGAL/Triangulation_2.h
+++ b/Triangulation_2/include/CGAL/Triangulation_2.h
@@ -2972,9 +2972,9 @@ exact_locate(const Point& p,
              Face_handle start) const
 #endif // no CGAL_NO_STRUCTURAL_FILTERING
 {
+  li = 4; //general init to avoid warnings.
+  lt = OUTSIDE_AFFINE_HULL; //same
   if (dimension() < 0) {
-    lt = OUTSIDE_AFFINE_HULL;
-    li = 4; // li should not be used in this case
     return Face_handle();
   }
   if( dimension() == 0) {
@@ -2983,10 +2983,6 @@ exact_locate(const Point& p,
     if (xy_equal(p,finite_vertex()->face()->vertex(0)->point())){
       lt = VERTEX ;
     }
-    else{
-      lt = OUTSIDE_AFFINE_HULL;
-    }
-    li = 4; // li should not be used in this case
     return Face_handle();
   }
   if(dimension() == 1){

